docs/doxygen.mk

```diff
@@ -999,6 +1040,36 @@ USE_HTAGS              = NO
 
 VERBATIM_HEADERS       = YES
 
+# If the CLANG_ASSISTED_PARSING tag is set to YES then doxygen will use the
+# clang parser (see: http://clang.llvm.org/) for more accurate parsing at the
+# cost of reduced performance. This can be particularly helpful with template
+# rich C++ code for which doxygen's built-in parser lacks the necessary type
+# information.
+# Note: The availability of this option depends on whether or not doxygen was
+# generated with the -Duse-libclang=ON option for CMake.
+# The default value is: NO.
+
+CLANG_ASSISTED_PARSING = NO
```

Comment:
We are not using it at the moment right. Why have it ?

This is causing warnings too.
```
warning: Tag `CLANG_ASSISTED_PARSING' at line 1052 of file `<path>/build/docs/doxygen.mk.out' belongs to an option that was not enabled at compile time.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u", or recompile doxygen with this feature enabled.
warning: Tag `CLANG_OPTIONS' at line 1060 of file `<path>/build/docs/doxygen.mk.out' belongs to an option that was not enabled at compile time.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u", or recompile doxygen with this feature enabled.
warning: Tag `CLANG_COMPILATION_DATABASE_PATH' at line 1071 of file `<path>/build/docs/doxygen.mk.out' belongs to an option that was not enabled at compile time.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u", or recompile doxygen with this feature enabled.
```

Reply:
Removing those lines throws a nice output for `make docs`

```
make docs
Generating Documentation
Built target docs
```
It is so peaceful to see no warnings from docs, :+1: @umar456

Reply:
Fixed.